releases/cosmwasm/2025-02-XRPL-v1.0.0.md

```diff
@@ -0,0 +1,364 @@
+XRPPL_MULTISIG_PROVER=$(cat ./axelar-chains-config/info/$ENV.json | jq ".axelar.contracts.XrplMultisigProver[\"$CHAIN\"].address" | tr -d '"')
+MULTISIG=$(cat ./axelar-chains-config/info/$ENV.json | jq .axelar.contracts.Multisig.address | tr -d '"')
+REWARDS=$(cat ./axelar-chains-config/info/$ENV.json | jq .axelar.contracts.Rewards.address | tr -d '"')
+```
+
+- Gov proposal environment variables. Update these for each network
+
+| Network              | `PROVER_ADMIN`                                  | `DEPOSIT_VALUE` | `REWARD_AMOUNT`     |
+| -------------------- | ----------------------------------------------- | --------------- | ------------------- |
+| **Devnet-amplifier** | `axelar1lsasewgqj7698e9a25v3c9kkzweee9cvejq5cs` | `100000000`     | `1000000uamplifier` |
+| **Stagenet**         | `axelar1l7vz4m5g92kvga050vk9ycjynywdlk4zhs07dv` | `100000000`     | `1000000uaxl`       |
+| **Testnet**          | `axelar17qafmnc4hrfa96cq37wg5l68sxh354pj6eky35` | `2000000000`    | `1000000uaxl`       |
+| **Mainnet**          | `axelar1pczf792wf3p3xssk4dmwfxrh6hcqnrjp70danj` | `2000000000`    | `1000000uaxl`       |
+
+```bash
+PROVER_ADMIN=[prover admin who is responsible for the contract's operations]
+DEPOSIT_VALUE=[deposit value]
+REWARD_AMOUNT=[reward amount]
+RUN_AS_ACCOUNT=[wasm deployer/governance address]
+```
+
+> **_NOTE:_** > `--runAs $RUN_AS_ACCOUNT` is only required for Devnet-amplifier. Do not use `--runAs` for Stagenet, Testnet, or Mainnet.
+
+5. Register `XRPLGateway` on the Router.
+
+> **_NOTE:_**
+> Add a community post for the Mainnet Proposal. e.g., https://www.mintscan.io/axelar/proposals/274
+
+```bash
+node cosmwasm/submit-proposal.js execute \
+  -c Router \
+  -t "Register Gateway for XRPL" \
+  -d "Register Gateway address for XRPL at Router contract" \
+  --runAs $RUN_AS_ACCOUNT \
+  --deposit $DEPOSIT_VALUE \
+  --msg "{
+    \"register_chain\": {
+      \"chain\": \"$CHAIN\",
+      \"gateway_address\": \"$GATEWAY\",
+      \"msg_id_format\": \"hex_tx_hash\"
+      }
+    }"
+```
+
+```bash
+axelard q wasm contract-state smart <router-address> '{"chain_info": "xrpl"}' --output json | jq .
+
+# You should see something like this:
+{
+  "data": {
+    "name": "xrpl",
+    "gateway": {
+      "address": "axelar1hzz0s0ucrhdp6tue2lxk3c03nj6f60qy463we7lgx0wudd72ctmsee8enx"
+    },
+    "frozen_status": 0,
+    "msg_id_format": "hex_tx_hash"
+  }
+}
+```
+
+6. Update `ampd` with the XRPL chain configuration. Verifiers should use their own `rippled` RPC node for the `chain_rpc_url` in production.
+
+| Network              | `http_url`                               |
+| -------------------- | ---------------------------------------- |
+| **Devnet-amplifier** | `https://s.devnet.rippletest.net:51234/` |
+| **Stagenet**         | `https://s.altnet.rippletest.net:51234/` |
+| **Testnet**          | `https://s.altnet.rippletest.net:51234/` |
```

Comment:
we will require any rate limits on these to be lifted for our verifier set. all requests from from a nat gateway.

Reply:
We haven't noticed any rate limiting being enforced so far for `devnet` and `testnet`. Will check with Ripple during our call today.